### PR TITLE
Relocate the imports to methods that require them

### PR DESCRIPTION
**Pull request by dgriff-tools**

### CODE REVIEW

File: signalfx/__init__.py [hunk 1]
```diff
@@ -109,6 +108,9 @@ def ingest(self, token, endpoint=None, timeout=None):
             timeout=timeout or self._timeout)
 
     def signalflow(self, token, endpoint=None, timeout=None):
+
+        import signalflow
```
Comment (mpetazzoni) [inline signalfx/__init__.py]: This breaks the method's documentation. Move to just before the `return`. Also don't need the empty lines around it.

**Comment by mpetazzoni** [comment]
I'd suggest doing the same thing for the `ingest()` and `rest()` methods, moving their respective sub-module import at the top of those methods. Don't forget to change your commit message/PR title accordingly.

Also, commit messages should follow the format described at http://chris.beams.io/posts/git-commit/

File: signalfx/__init__.py (hunk 1 above)
Comment (dgriff-tools) [inline signalfx/__init__.py]: fixing that now

**Comment by mpetazzoni** [comment]
Cool; looks much better. Thanks for the cleanup and quick turnaround.

**Pull request merged**